lib/tessel/deploy.js

```diff
@@ -193,39 +201,31 @@ Tessel.prototype.restartScript = function(opts) {
       .then(() => {
         if (isPush) {
           // Start the script from flash memory
-          return actions.startPushedScript(this, opts.entryPoint, opts)
+          return exportables.start(this, opts.entryPoint, opts)
             .then(resolve).catch(reject);
         } else {
           // Start the script in RAM
-          return actions.runScript(this, filepath, opts)
+          return exportables.run(this, opts)
             .then(resolve).catch(reject);
         }
       })
       .catch((error) => {
-        if (error.message.indexOf('No such file or directory') !== -1) {
-          error = '"' + opts.entryPoint + '" not found on ' + this.name;
+        if (error.message.includes('No such file or directory')) {
+          error = `"${opts.entryPoint}" not found on ${this.name}`;
         }
 
         return reject(error);
       });
   });
 };
 
-actions.execRemoteCommand = function(tessel, command, filepath) {
-  return tessel.simpleExec(commands[command](filepath))
-    .catch(function(err) {
-      if (err.length > 0) {
-        throw new Error('we had a simple exec error!', err);
-      }
-    });
-};
 
-actions.findProject = function(opts) {
-  return new Promise(function(resolve, reject) {
+exportables.findProject = function(opts) {
```

Comment:
Must be called with an `opts` object that has already resolved the language to use. This is necessary for locating language/runtime specific files and configurations